src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -52,6 +52,7 @@ internal class HttpManager : IHttpManager
             bool doNotThrow,
             bool retry,
             X509Certificate2 bindingCertificate,
+            Func<HttpClient> validateServerCertificateCallback,
```

Comment:
Why do you need a callback instead of an HttpClient? smth like "HttpClient customHttpClient"

The Func is always invoked, so the object is constructed anyway (besides, for lazy optimizations recommendation is to use Lazy<T>)